x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/BulkUpdateApiKeyResponse.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.apikey;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class BulkUpdateApiKeyResponse extends ActionResponse implements ToXContentObject, Writeable {
+
+    private final List<String> updated;
+    private final List<String> noops;
+    private final Map<String, Exception> errorDetails;
+
+    public BulkUpdateApiKeyResponse(final List<String> updated, final List<String> noops, final Map<String, Exception> errorDetails) {
+        this.updated = updated;
+        this.noops = noops;
+        this.errorDetails = errorDetails;
+    }
+
+    public BulkUpdateApiKeyResponse(StreamInput in) throws IOException {
+        super(in);
+        this.updated = in.readStringList();
+        this.noops = in.readStringList();
+        this.errorDetails = in.readMap(StreamInput::readString, StreamInput::readException);
+    }
+
+    public List<String> getUpdated() {
+        return updated;
+    }
+
+    public List<String> getNoops() {
+        return noops;
+    }
+
+    public Map<String, Exception> getErrorDetails() {
+        return errorDetails;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject().stringListField("updated", updated).stringListField("noops", noops);
+        builder.startObject("errors");
```

Comment:
I'm fine with that. I went with the structure from the proposal where we state:

> Detailed error messages. It will not present if `errors.count` is 0.

I interpreted this to mean always display the error count. No strong preference either way.